### PR TITLE
Set team members to be unknown/confirmed

### LINKED ISSUE #113: [Model & Logic] Set team members to be unknown/accepted

### LINKED ISSUE #150: [Tests] Write tests for confirm-member

### PR DESCRIPTION
**Pull request by yuhinarita**
Closes #113 
Tests to be written later in #150

Edit: Wrote tests, now it also closes #150

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ConfirmMemberCommand.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        this.groupName = groupName;
+        this.queryableCourseMateSet = queryableCourseMateSet;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        Set<List<CourseMate>> courseMateSet;
+        List<QueryableCourseMate> queryableCourseMates = new ArrayList<>();
+        for (QueryableCourseMate q : queryableCourseMateSet) {
```
Comment (benson1029) [inline src/main/java/seedu/address/logic/commands/ConfirmMemberCommand.java:60]: Consider using `addAll` here?

File: src/main/java/seedu/address/logic/commands/ConfirmMemberCommand.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        } catch (GroupNotFoundException e) {
+            throw new CommandException(Messages.MESSAGE_INVALID_GROUP_NAME);
+        }
+
+        Group modifiedGroup = new Group(toModify);
+
+        List<CourseMate> courseMateList = new ArrayList<>();
+        int index = 0;
+        for (List<CourseMate> courseMateConfirmList: courseMateSet) {
+            //If there are more than 1 matching name
```
Comment (benson1029) [inline src/main/java/seedu/address/logic/commands/ConfirmMemberCommand.java:85]:
```suggestion
            // If there are more than 1 matching name
```

File: src/main/java/seedu/address/logic/commands/UnconfirmMemberCommand.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        this.groupName = groupName;
+        this.queryableCourseMateSet = queryableCourseMateSet;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        Set<List<CourseMate>> courseMateSet;
+        List<QueryableCourseMate> queryableCourseMates = new ArrayList<>();
+        for (QueryableCourseMate q : queryableCourseMateSet) {
```
Comment (benson1029) [inline src/main/java/seedu/address/logic/commands/UnconfirmMemberCommand.java:60]: same here

File: src/main/java/seedu/address/logic/commands/UnconfirmMemberCommand.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+            courseMateList.add(courseMateUnconfirmList.get(0));
+            index += 1;
+        }
+
+        try {
+            for (CourseMate courseMate: courseMateList) {
+                CourseMate targetCourseMate = modifiedGroup.findCourseMate(courseMate.getName()).get(0);
+                CourseMate editedCourseMate = unconfirmCourseMate(courseMate);
+                modifiedGroup.setCourseMate(targetCourseMate, editedCourseMate);
+            }
```
Comment (benson1029) [inline src/main/java/seedu/address/logic/commands/UnconfirmMemberCommand.java:99]: looks like there is quite a bit of code duplication. consider abstracting it out?

Comment (benson1029) [reply src/main/java/seedu/address/logic/commands/UnconfirmMemberCommand.java:99]: for example, OOP it a bit more by creating an abstract class and both ConfirmMember and UnconfirmMember inherits from it

File: src/main/java/seedu/address/logic/parser/ConfirmMemberCommandParser.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        requireNonNull(args);
+        ArgumentMultimap argMultiMap = ArgumentTokenizer.tokenize(args, PREFIX_COURSEMATE);
+
+        try {
+            Name name = ParserUtil.parseName(argMultiMap.getPreamble());
+            Set<QueryableCourseMate> queryableCourseMateSet =
+                    ParserUtil.parseQueryableCourseMates(argMultiMap.getAllValues(PREFIX_COURSEMATE));
+
+            // will get caught by the catch clause, leave empty.
+            if (queryableCourseMateSet.isEmpty()) {
```
Comment (benson1029) [inline src/main/java/seedu/address/logic/parser/ConfirmMemberCommandParser.java:34]: Discouraged to use try-catch for control flow.

**Review by benson1029 — APPROVED** [review]
Optional to improve the design a bit.

File: src/main/java/seedu/address/logic/parser/ConfirmMemberCommandParser.java (hunk 5 above)
Comment (yuhinarita) [reply src/main/java/seedu/address/logic/parser/ConfirmMemberCommandParser.java:34]: Ok I'll refactor this. I got this idea from `AddMemberCommand` which uses the exact same logic here, so I'lll try modify that as well.

File: src/main/java/seedu/address/logic/commands/UnconfirmMemberCommand.java [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+
+        Group toModify;
+        try {
+            toModify = model.findGroup(groupName);
+        } catch (GroupNotFoundException e) {
+            throw new CommandException(Messages.MESSAGE_INVALID_GROUP_NAME);
+        }
+
+        Group modifiedGroup = new Group(toModify);
+        System.out.println("init Group: " + modifiedGroup.asUnmodifiableObservableList());
```
Comment (yuhinarita) [inline src/main/java/seedu/address/logic/commands/UnconfirmMemberCommand.java:81]: Forgot to remove debug print

File: src/main/java/seedu/address/logic/commands/UnconfirmMemberCommand.java (hunk 4 above)
Comment (yuhinarita) [reply src/main/java/seedu/address/logic/commands/UnconfirmMemberCommand.java:99]: ok thanks!

**Pull request merged**